src/Features/Core/Portable/ConvertTupleToStruct/AbstractConvertTupleToStructCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,865 @@
+                    continue;
+                }
+
+                if (AreEquivalent(comparer, tupleType, childType))
+                {
+                    changed = true;
+                    ReplaceWithObjectCreation(
+                        syntaxFacts, editor, typeName, typeParameters, 
+                        qualifiedTypeName, startingNode, childCreation);
+                }
+            }
+
+            return changed;
+        }
+
+        private static bool AreEquivalent(StringComparer comparer, INamedTypeSymbol tupleType, INamedTypeSymbol childType)
+            => SymbolEquivalenceComparer.Instance.Equals(tupleType, childType) &&
+               NamesMatch(comparer, tupleType.TupleElements, childType.TupleElements);
+
+        private static bool NamesMatch(
+            StringComparer comparer, ImmutableArray<IFieldSymbol> fields1, ImmutableArray<IFieldSymbol> fields2)
+        {
+            if (fields1.Length != fields2.Length)
+            {
+                return false;
+            }
+
+            for (var i = 0; i < fields1.Length; i++)
+            {
+                if (!comparer.Equals(fields1[i].Name, fields2[i].Name))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private void ReplaceWithObjectCreation(
+            ISyntaxFactsService syntaxFacts, SyntaxEditor editor, string typeName, ImmutableArray<ITypeParameterSymbol> typeParameters,
+            TNameSyntax qualifiedTypeName, SyntaxNode startingCreationNode, TTupleExpressionSyntax childCreation)
+        {
+            // Use the callback form as anonymous types may be nested, and we want to
+            // properly replace them even in that case.
+            editor.ReplaceNode(
+                childCreation,
+                (currentNode, g) =>
+                {
+                    var currentTupleExpr = (TTupleExpressionSyntax)currentNode;
+
+                    // If we hit the node the user started on, then add the rename annotation here.
+                    var typeNameNode = startingCreationNode == childCreation
+                        ? CreateStructNameNode(g, typeName, typeParameters, addRenameAnnotation: true)
+                        : qualifiedTypeName;
+
+                    syntaxFacts.GetPartsOfTupleExpression<TArgumentSyntax>(
+                        currentTupleExpr, out var openParen, out var arguments, out var closeParen);
+                    arguments = ConvertArguments(syntaxFacts, g, arguments);
+
+                    return CreateObjectCreationExpression(typeNameNode, openParen, arguments, closeParen)
+                        .WithAdditionalAnnotations(Formatter.Annotation);
+                });
+        }
+
+        private SeparatedSyntaxList<TArgumentSyntax> ConvertArguments(ISyntaxFactsService syntaxFacts, SyntaxGenerator generator, SeparatedSyntaxList<TArgumentSyntax> arguments)
```

Comment:
>ConvertArguments [](start = 53, length = 16)

One of the two `ConvertArguments` overloads may be unused.

Reply:
both are used.  :)

Reply:
the second is called by the first.